src/DeploymentProvider/AzureSpringCloudDeploymentProvider.ts

```diff
@@ -0,0 +1,207 @@
+import * as core from '@actions/core';
+import { v4 as uuidv4 } from 'uuid';
+import { Package, PackageType } from 'azure-actions-utility/packageUtility';
+import { Actions, ActionParameters, ActionParametersUtility } from '../operations/actionParameters';
+import { AppPlatformManagementClient, AppPlatformManagementModels as Models } from '@azure/arm-appplatform'
+import { getDefaultAzureCredential } from '@azure/identity'
+import { DeploymentHelper as dh } from "./DeploymentHelper";
+import tar = require('tar');
+
+export class AzureSpringCloudDeploymentProvider {
+
+    defaultInactiveDeploymentName = 'staging';
+
+    params: ActionParameters;
+    client: AppPlatformManagementClient;
+
+    constructor() {
+        this.params = ActionParametersUtility.getParameters();
+    }
+
+    public async PreDeploymentStep() {
+        const token = getDefaultAzureCredential();
+        this.client = new AppPlatformManagementClient(token, this.params.AzureSubscription);
+        const serviceList = await this.client.services.listBySubscription();
+        let filteredResources: Array<Models.ServiceResource> = [];
+        serviceList.forEach(service => {
+            if(service.name == this.params.ServiceName) {
+                filteredResources.push(service);
+            }
+        });
+        if (!filteredResources || filteredResources.length == 0) {
+            throw new Error('ResourceDoesntExist' + this.params.ServiceName);
+        }
+        else if (filteredResources.length == 1) {
+            const beginStr = '/resourceGroups/';
```

Comment:
Resource ID is case insensitive. We should not assume it will be `/resourceGroups/`, it may be `/resourcegroups/` or other cases.

Same below.

Reply:
Consider using a regex, it will be much more trivial:

```javascript
var reg = new RegExp('/subscriptions/([^/]+)/resourceGroups/([^/]+)/providers/Microsoft.AppPlatform/Spring/([^/]+)', 'i')
var match = '<resource-id-to-be-matched>`.match(reg)

// now check if match is null
// if not, it will be an array like object, with 4 elements: 0) the resource ID, 1) subscription, 2) resource group, 3) resource name.
```